### ISSUE TITLE
Perldoc is broken!

### ISSUE THREAD

**Issue by ANSI-C** [OpusVL/perldoc.perl.org#36]
Search feature is gone, links do not work, the new interface is space wasting and badly formatted, can we have back the good old site? The new one is just, sorry to be rude, horrible and unusable.

**Comment by JohnMica** [comment]
@ANSI-C 

- search (coming in the next iteration)

- links dont work - care to elaborate what links are you talking about ? what pages are the ones causing this links not to work (besides the ones already presented in other issues #35 , #28 ) ?

- badly formatted = care to elaborate what you actually mean by this ? 

without you being specific in your 'issue descriptions' it is not extremely useful to the iterative process of improving the new site.

**Comment by ANSI-C** [comment]
> 
> 
> @ANSI-C
> 
>     * search (coming in the next iteration)
> 
>     * links dont work - care to elaborate what links are you talking about ? what pages are the ones causing this links not to work (besides the ones already presented in other issues #35 , #28 ) ?
> 
>     * badly formatted = care to elaborate what you actually mean by this ?
> 
> 
> without you being specific in your 'issue descriptions' it is not extremely useful to the iterative process of improving the new site.

sorry for being too coincided, but was shocked by realizing that perldoc has been completely changed without testing it properly beforehand and (AFAIK) without user feedback on new design. About bugs:

- just an example in page https://perldoc.perl.org/5.30.0/functions/pack.html you have the _unpack_ link not working, with anchor set to _https://perldoc.perl.org/5.30.0/functions/pack.html#unpack-TEMPLATE%2cEXPR_

- about bad formatting, the new style is just utterly space wasting, just compare with same browser _https://perldoc.pl/functions/pack_ and _https://perldoc.perl.org/5.30.0/functions/pack.html_, c'mon...

**Comment by brtastic** [comment]
I agree with ANSI-C on his criticism. I feel like browsing some bootstrap-based front page of programming language and not a technical documentation. The fonts seem too big, there's a lot of pixels wasted for either blank space (left, right margin, top bar) or things that tell you nothing about what you're looking for (Perl 5 version 30.0 documentation being h1...). When you add current lack of searching on the page it becomes very hard to have good impressions of the current layout. Even the beta version info seems too big.
I just feel like most of my screen is now pure visual noise. I don't want it to be pretty, I want it readable, but right now it's just not, not for me at least.
![Screenshot from 2019-06-17 23-15-16](https://user-images.githubusercontent.com/44323413/59637340-d39c6300-9155-11e9-8519-136334eb37a8.png)
Look at this as an example. 
- About a quarter of my screen's height is gone for top bar and it's margin. While it's nice for a front page, it could be smaller when reading a docs page
- Breadcrumbs seem OK, but they could be a little bit smaller than regular text
- H1 perl version - visual noise. It's not what I was looking for
- name of the function - bless. The size is good but it looks like it's wrapped part of previous line. It doesn't look as if it's the main topic of this page. It could use a horizontal line just beneath it (for example)
- Categories - same as breadcrumbs. Also, for me they could as well be the last thing on this page, not one of the first ones
- usage, function signatures - this looks awful, the padding should be much smaller and it could use something like very subtle border or a bit darker background
- the rest of the page looks quite nice, but I don't like the fact that it doesn't use full width and doesn't border itself from the white background
- top bar and beta version badge are making it extra difficult to use this website by hiding my scroll bar (I know it's a bug)
While it would be nice to see a good looking perl docs website, this version needs a lot of work to be at least equally pleasant to use as the previous, old one.

**Comment by brtastic** [comment]
To add some positive input I've made quick-and-dirty console editing to make it look more pleasing to my own eye. Of course I didn't change everything listed above, but this is what I ended up with
![Screenshot from 2019-06-18 00-00-57](https://user-images.githubusercontent.com/44323413/59639782-748e1c80-915c-11e9-8765-b6afeecdfd3c.png)
It's much quicker to find what I'm looking for here, and it wasn't much work. Also the entropy is exactly the same (nothing was added or removed, only moved around)

**Comment by hakonhagland** [comment]
Thanks for working on the new site. But I have to agree with @ANSI-C that there should be a way to access (a link ?) the old site (that was working perfectly in my browser) while bugs for the new site are being fixed. Since I cannot scroll on the new site (see issue #16) I cannot use it. Also,  when I google for a topic on perldoc.perl.org only the new page comes up and it is frustrating. Why not test out the beta version on a different URL until most bugs were fixed before releasing it?

**Comment by ANSI-C** [comment]
And _last but not least_, I would like to have good old sidebar back, with quick links to core modules list, functions, and so on. We have one third of page width just blank, you have all the vertical space to add back the sidebar, even with responsive layout if you like.

**Comment by ANSI-C** [comment]
For anyone interested you can download the **last copy of old site** for offline view [HERE](http://web.archive.org/web/20170709212848/http://perldoc.perl.org/perldoc-html.tar.gz)

**Comment by randalmacdonald** [comment]
As a long time Perl developer, I appreciate the effort to bring the language's web-based documentation up to date.  However, not providing search functionality in this beta version we're all now apparently forced to use is unacceptable.  Lack of site search has made my Perl programming work much more cumbersome and time-consuming.  When will search become available?

(Thanks to ANSI-C for recommending the temporary work-around of downloading the Perl 5.22 tarball, which I've done and which allows me to search that version's documentation.  Kudos indeed to the creator(s) of that site's Javascript search engine!)

I agree with the other comments posted here by ANSI-C, brtastic and hakonhagland.  I particularly want to echo what hakonhagland asked:  "Why not test out the beta version on a different URL until most bugs were fixed before releasing it?"

**Comment by Grinnz** [comment]
As mentioned elsewhere, for those who need it in the meantime, the stable alternative https://perldoc.pl has a search feature.

**Comment by ANSI-C** [comment]
> 
> 
> As mentioned elsewhere, for those who need it in the meantime, the stable alternative https://perldoc.pl has a search feature.

your site is also very useful, thanks!

**Comment by nuclight** [comment]
https://perldoc.pl is banned in some countries, at least Russia (governments are blocking entire segments in war with e.g. Telegram, many innocent sites are also affected).

**Comment by ANSI-C** [comment]
> 
> 
> https://perldoc.pl is banned in some countries, at least Russia (governments are blocking entire segments in war with e.g. Telegram, many innocent sites are also affected).

Then the best option is to download tarball of old site and use it offline, I am already doing that, and it works even better than old online site:
[http://web.archive.org/web/20170709212848/http://perldoc.perl.org/perldoc-html.tar.gz](http://web.archive.org/web/20170709212848/http://perldoc.perl.org/perldoc-html.tar.gz)

**Comment by harmic** [comment]
Another option for those that find the new perldoc unusable: https://devdocs.io/

That site supports perl fairly well, supports offline usage, and has good search capabilities.

**Comment by jaypittman** [comment]
Looking here:
https://perldoc.perl.org/5.30.0/index-functions-by-cat.html#Numeric-functions
This defines exp() as: raise I to a power
Raise what? Is this a one? Or an ell? Did someone mean e?

**Comment by Grinnz** [comment]
Yes, that is a misrendering of an italicized `e`. You can see the same issue in the description of `__SUB__`.
